mypy/meet.py

```diff
@@ -25,9 +25,19 @@ def meet_types(s: Type, t: Type) -> Type:
     return t.accept(TypeMeetVisitor(s))
 
 
-def meet_simple(s: Type, t: Type, default_right: bool = True) -> Type:
+def meet_simple(s: Type, t: Type) -> Type:
+    """Return the type s "narrowed down" to t.
+
+    Note that this is not symmetric with respect to s and t.
+
+    TODO: Explain what this does in more detail and how this is
+          different from meet_types.
+    """
     if s == t:
         return s
+    if isinstance(t, AnyType):
```

Comment:
Without this, `testUnionAnyIsInstance` will fail -- the type in the else block will be `Union[int, Any]`, though it should be `Any`. This makes is possible to narrow down from `Union[int, Any]` to `Any`, and this is the only place where this code is used. Previously this worked (probably accidentally) because `Union[int, Any]` got simplified to just `Any`. This won't have any other effect, as far I can see.